db/database.go

```diff
@@ -881,6 +883,18 @@ func (db *Database) processForEachDocIDResults(callback ForEachDocIDFunc, limit
 		}
 
 	}
+
+	// Have to iterate over any remaining results to clear the reader
+	// Otherwise we get "the result must be closed before accessing the meta-data" on close details on CBG-1666
+	if !endOfResultsReached && !db.UseViews() {
+		for {
+			found := results.Next(nil)
+			if !found {
+				break
+			}
+		}
```

Comment:
Not sure how much the compiler will optimise away in your case, but this still feels cleaner even if it makes no performance difference.

```suggestion
		for r.Next(nil) {
			// noop to drain results
		}
```